lib/app/services/api.dart

```diff
@@ -44,6 +48,67 @@ class ApiService {
     }).toList();
   }
 
+  Future<List<Course>> listCourses() async {
+    var response = await network.get('courses?');
+
+    var body = json.decode(response.body);
+
+    return Future.wait((body['data'] as List<dynamic>).map((data) async {
+      data = data as Map<String, dynamic>;
```

Comment:
Instead of rebinding the identifier to a potentially different type, you could cast the iterable before mapping.